aiida/cmdline/commands/cmd_import.py

```diff
@@ -47,33 +50,44 @@ def _echo_error(  # pylint: disable=unused-argument
     :type more_archives: bool
     :param raised_exception: Exception raised during error.
     :type raised_exception: `Exception`
+    :param fail_fast: Exit the import/migration immediately if an error occurs.
+    :type fail_fast: bool
     """
     from aiida.tools.importexport import close_progress_bar, IMPORT_LOGGER
 
+    global RETURN_NON_ZERO
```

Comment:
Aaahhh a global variable!! :-D

Do you really need it? Essentially you want to check if the _echo_error was ever called?
Then just do it in the caller, no? Any call to _echo_error should return a boolean flag 'return_non_zero' and you pass it up to the relevant function?

Reply:
Yeah. It was the quickest (and dirtiest) way to do it 😅 

I'll take a look at rewriting it.

Reply:
Alright, I've rewritten it, but the code is now much more unclear I think.
The issue here is that it should retain the information if it is set to `True`, but still allow for trying to import more archives, if supplied.

Reply:
Agree it's now a bit more involved, but the global variable would probably get you into trouble when testing (because it's run more than once from the same interpreter instance).

Reply:
> Agree it's now a bit more involved, but the global variable would probably get you into trouble when testing (because it's run more than once from the same interpreter instance).

... which is why I shouldn't wait with writing up the tests I guess 😅